parser/impl/parser.cpp

```diff
@@ -240,28 +240,123 @@ sPtr<ast::RecordType> Parser::parseRecordType() {
 }
 
 // ---- @CrazyDream1
-
-sPtr<ast::Variable> Parser::parseVariable() {
-    auto token = m_lexer.Next();
-    if (token.type != TokenType::Var) {
-        m_errors.push_back(Error{
-            .pos = token.pos,
-            .message = "Expected \"var\" keyword but didn't find it.",
-        });
+  
+    sPtr<ast::Variable> Parser::parseVariable() {
+	    ast::Variable variable;
+        Token token = skipWhile(isNewLine);
+        if (token.type != TokenType::Var) {
+            m_errors.push_back(Error{
+                .pos = token.pos,
+                .message = "Expected \"var\" keyword but didn't find it.",
+            });
+        }
+        token = skipWhile(isNewLine);
+        if (token.type != TokenType::Identifier) {
+            m_errors.push_back(Error{
+                .pos = token.pos,
+                .message = "Expected an identifier after \"var\" keyword.",
+            });
+        }
+	    variable.name = token;
+        token = skipWhile(isNewLine);
+        if (token.type != TokenType::Is && token.type != TokenType::Colon) {
+            m_errors.push_back(Error{
+                .pos = token.pos,
+                .message = "Expected an \"is\" keyword or : after the identifier.",
+            });
+        }
+        switch (token.type)
+        {
+	        case TokenType::Colon:
+                variable.type = parseType();
+                if ((token = m_lexer.Peek()).type != TokenType::Is) break;
+		        else m_lexer.Next();
+            case TokenType::Is:
+                variable.expression = parseExpression();
+                break;
+            default:
```

Comment:
Don't really need a `switch` here, and it impairs readability. Since a return type is optional, first check (with a normal `if` condition) if there is a colon and so parse the type. Then just continue normally to checking for `is`. After that, you can report an error if both `variable.type` and `variable.expression` are null